react-renderer/src/inspector/views/mod/shapedefs/arrow.json

```diff
@@ -0,0 +1,55 @@
+{
```

Comment:
Great to finally have shapedefs in the frontend! We will eventually move the backend shapedefs into the frontend as well. A more principled way to generate these mod widgets: each shapedef will also include the datatype for each property. We can then derive the `inputType` based on it (e.g. `FloatT` -> `range`, `StringT` -> `select'). The new shapedefs should also define the domain of their values (e.g. `arrowheadStyle` -> `["arrowhead-1", "arrowhead-2"]`). WIth this information, we can generate sample functions for resampling shapes as well.

Reply:
@wodeni I did consider that, but I opted for extra customization, in case someone thinks a `number` input is more appropriate than `range` for their custom property of type `FloatT`. Also, I think the same types can map to multiple input options; e.g, the option to edit the content of a `text` should be presented as a text field, but `arrowheadStyle` should be a select, though both have values of type `String`. But I think once the backend shapedefs are moved to the front end it could be possible to do what you're suggesting.

Reply:
Good point. My current thought is to annotate shape properties with types and a range of values, and then make the decision on how to present them. Maybe an extension point within vizdoc to override defaults will give users enough flexibility.